test/trento/software_updates/discovery_test.exs

```diff
@@ -252,6 +256,74 @@ defmodule Trento.SoftwareUpdates.DiscoveryTest do
     end
   end
 
+  defp fail_on_getting_system_id(fully_qualified_domain_name, discovery_error) do
```

Comment:
Maybe better put this private helper functions at the end of the file, instead of the middle?